### ISSUE TITLE
Cannot use Substitute in prjconf when building Debian packages

### ISSUE THREAD

**Issue by bluca** [openSUSE/obs-build#322]
When using a Substitute: foo bar in the prjconfig the OBS resolver will happily select bar and start a build, but then dpkg-buildpackage fails immediately as it checks for its original build-dependencies and finds that foo is missing.

build-recipe-rpm calls the substitutedeps script which I believe mangles the spec to make the problem go away (I'm not the world foremost Perl expert so I might be wrong):

https://github.com/openSUSE/obs-build/blob/master/build-recipe-spec#L50
https://github.com/openSUSE/obs-build/blob/master/substitutedeps

A quick and easy solution would be to add to the dpkg-buildpackage call in build-recipe-spec:

https://github.com/openSUSE/obs-build/blob/master/build-recipe-dsc#L98

The "-d" option, which disabled the build-dependency check. After all this is already done by the resolver before starting a build.

Would this be a reasonable and acceptable fix? Is there an alternative?

Thank you!
